### PR TITLE
fix: update Discord ID regex to include 19 digit IDs

### PR DESCRIPTION
**Pull request by frank-cywong**
#### Description
From July 22 onwards, new Discord accounts will have 19 digit Discord IDs (https://snowsta.mp/?l=en-us&z=6&f=atx10uuark-7ps), this is similar to #971 in terms of how it changes the regex.

#### Screenshot (if UI-related)
NA

#### To-Dos
NA

#### Issues Fixed or Closed
No such issue reported yet as the bug could only occur after July 22 at the earliest

### CODE REVIEW

File: src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const UserGeneralSettings: React.FC = () => {
   const UserGeneralSettingsSchema = Yup.object().shape({
     discordId: Yup.string()
       .nullable()
-      .matches(/^\d{17,18}$/, intl.formatMessage(messages.validationDiscordId)),
+      .matches(/^\d{17,18,19}$/, intl.formatMessage(messages.validationDiscordId)),
```
Comment (TheCatLady) [inline src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx:83]:
```suggestion
      .matches(/^\d{17,19}$/, intl.formatMessage(messages.validationDiscordId)),
```

File: src/components/UserProfile/UserSettings/UserNotificationSettings/UserNotificationsDiscord.tsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ const UserNotificationsDiscord: React.FC = () => {
           .required(intl.formatMessage(messages.validationDiscordId)),
         otherwise: Yup.string().nullable(),
       })
-      .matches(/^\d{17,18}$/, intl.formatMessage(messages.validationDiscordId)),
+      .matches(/^\d{17,18,19}$/, intl.formatMessage(messages.validationDiscordId)),
```
Comment (TheCatLady) [inline src/components/UserProfile/UserSettings/UserNotificationSettings/UserNotificationsDiscord.tsx:49]:
```suggestion
      .matches(/^\d{17,19}$/, intl.formatMessage(messages.validationDiscordId)),
```

**Review by TheCatLady — CHANGES_REQUESTED** [review]
Please test your changes next time. 😉

**Comment by frank-cywong** [comment]
Right, thanks for the heads up, probably shouldn't have deleted the entire todo list part of the PR

**Review by sct — APPROVED** [review]

**Comment by sct** [comment]
Thanks for fixing it up!

**Comment by sct** [comment]
@all-contributors please add @frank-cywong for code

**Pull request merged**